### PR TITLE
Cut new release 3.11.0

### PR DESCRIPTION
**Pull request by Spaak**
As per the checklist: https://github.com/pymc-devs/pymc3/wiki/Release-Checklist .

### CODE REVIEW

**Review by OriolAbril — APPROVED** [review]

File: RELEASE-NOTES.md [hunk 1]
```diff
@@ -14,6 +13,7 @@ It also brings some dreadfully awaited fixes, so be sure to go through the chang
   - 0-length dimensions are now ruled illegal for random variables and raise a `ValueError`.
 - In `sample_prior_predictive` the `vars` kwarg was removed in favor of `var_names` (see [#4327](https://github.com/pymc-devs/pymc3/pull/4327)).
 - Removed `theanof.set_theano_config` because it illegally changed Theano's internal state (see [#4329](https://github.com/pymc-devs/pymc3/pull/4329)).
+- We now depend on `Theano-PyMC` version `1.1.0` or greater (see [#4405](https://github.com/pymc-devs/pymc3/pull/4405)).
```
Comment (michaelosthege) [inline RELEASE-NOTES.md]: To be precise, this release depends on exactly `Theano-PyMC v1.1.0`.

Comment (michaelosthege) [reply RELEASE-NOTES.md]: Let's also add a line (above or in "Breaking changes") about the most important changes in `Theano-PyMC`:
```
Major refactorings were done in Theano-PyMC 1.1.0, if you implement custom `Op`s or interact with Theano in any way yourself, make sure to read the [Theano-PyMC 1.1.0 release notes](https://github.com/pymc-devs/Theano-PyMC/releases/tag/rel-1.1.0).
```

**Review by michaelosthege — CHANGES_REQUESTED** [review]
Just some things still missing from the readme

**Review by twiecki — APPROVED** [review]

**Comment by aloctavodia** [comment]
After merging #4310 we should add something like 

BART: Add a new `predict` method to compute out of sample predictions evaluated at new X values (see [#4310](https://github.com/pymc-devs/pymc3/pull/4310)).

**Comment by Spaak** [comment]
@aloctavodia ideally add the changes to the release notes inside #4310 before merging.

**Comment by michaelosthege** [comment]
> @aloctavodia ideally add the changes to the release notes inside #4310 before merging.

I just merged already. Editing the release notes in #4310 would have created a git conflict and also would have taken another 90 minutes of CI...

**Comment by twiecki** [comment]
Alright! All other issues are closed so we can merge this and release.

**Comment by Spaak** [comment]
Yep! On it right now.

**Review by michaelosthege — APPROVED** [review]
LGTM 🎉

**Comment by twiecki** [comment]
Woohoo!

**Pull request merged**